File: code/Quack.java
interface Duck {
   public void quack();
}

class Mallard implements Duck {
  public void quack() {
    System.out.println("QUACK!");
  }
}

class Pekin implements Duck {
  public void quack() {
    System.out.println("quack.");
  }
}

public class Quack {
  public static void main(String arg[]) {
    Duck[] ducks = {new Mallard(), new Pekin()};
    for (int i = 0; i < ducks.length; i++) {
      ducks[i].quack();
    }
  }
}

File: code/quack.go
package main

import "fmt"

type Duck interface {
  Quack()
}

type Mallard struct{}
type Pekin struct{}

func (m Mallard) Quack() {
  fmt.Println("QUACK!")
}

func (p Pekin) Quack() {
  fmt.Println("quack.")
}

func main() {
  ducks := []Duck{Mallard{}, Pekin{}}
  for _, duck := range ducks {
    duck.Quack()
  }
}
